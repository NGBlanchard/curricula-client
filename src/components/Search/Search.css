.search-form {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  padding-top: 10px;
  padding-bottom: 10px;
  margin-bottom: 20px;
  /* background-color: rgba(202, 80, 255, 0.19) */
}

#search-button {
  border: 1px solid #aaa;
  background-color: ivory;
  font-size: 1em;
  border-radius: 4px;
  padding: 6px 10px;
  font-weight: lighter;
  margin: 10px 0px 10px 0px;
 
}

#search-button:hover {
  cursor: pointer;
  box-shadow: 0 4px 8px 0 rgba(202, 80, 255, 0.19);
}

#searchTerm {
  border: 1px solid #ddd;
  font-size: 1em;
  margin: 10px 0px 10px 0px;
  padding: 10px;
  text-align: left;
  width: 300px;
  border-radius: 4px;
}

.topic-sort {
  background-color: #f7f8f9;
  color: #444;
  margin-left: 10px;
  height: 20px;;
}

.topic-sort:hover {
  box-shadow: 0 4px 7px 0 rgba(202, 80, 255, 0.19);
}

.topic-sort-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
}
