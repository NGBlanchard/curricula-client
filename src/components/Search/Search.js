import React from 'react';
import { Button, Input } from '../../components/Utils/Utils'

import './Search.css';


class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
    }
  }

  onChange = e => {
    this.setState({ [e.target.id]: e.target.value });
    };
  

  render() {
    return (
        <form onSubmit={this.onSearch} className="search-form">
            <label htmlFor="course-searchTerm"></label>
            <Input 
                type="text" 
                id="searchTerm"
                onChange={this.onChange}
                value={this.state.searchTerm} 
                placeholder="Hannah Arendt & The Walking Dead" 
                required 
            />
            <Button id="search-button" type="submit">Search Courses</Button>
        </form>
    );
  }
}

export default Search;