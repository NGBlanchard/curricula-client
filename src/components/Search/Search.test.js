import Search from "./Search";
import { BrowserRouter as Router } from "react-router-dom";
import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<Search />", () => {
  const courseList = [];
  const filterList = () => {
    return;
  };
  it("Renders without crashing", () => {
    mount(
      <Router>
        <Search courseList={courseList} filterList={() => filterList()} />
      </Router>
    );
  });
  it("renders Search component by default", () => {
    const wrapper = mount(
      <Router>
        <Search courseList={courseList} filterList={() => filterList()} />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
