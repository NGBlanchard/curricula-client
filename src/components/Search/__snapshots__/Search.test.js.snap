// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders Search component by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Search
      courseList={Array []}
      filterList={[Function]}
    >
      <div
        className="topic-sort-container"
      >
        <p>
          Sort by Course Topic: 
        </p>
        <select
          className="topic-sort"
          onChange={[Function]}
        >
          <option
            className="deselect-topic"
            key="none"
            value=""
          >
            None
          </option>
        </select>
      </div>
    </Search>
  </Router>
</BrowserRouter>
`;
